CMakeLists.txt

```diff
@@ -1,6 +1,30 @@
+cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+
+if(POLICY CMP0025)
+  cmake_policy(SET CMP0025 NEW)
+endif()
+
 project(MONGO_CXX_DRIVER LANGUAGES CXX)
 
-cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9")
```

Comment:
Yeah. I feel a _little_ bad about excluding gcc 4.8. We could possibly ifdef it and lower? I think it is only used in one place. But, lets set the bar high and see if we get pushback.